### ISSUE TITLE
FMU export -> "absoluteValue" to "relativeQuantity" 

### ISSUE THREAD

**Issue by BasilioPV** [OpenModelica/OpenModelica#12533]
### Description

In a model I have defined a temperature difference variable:

```
Modelica.Units.SI.TemperatureDifference Tdiff;
```
where:

```
type TemperatureDifference = Real (
    final quantity="ThermodynamicTemperature",
    final unit="K") annotation(absoluteValue=false);
```
and from Modelica Language Specification:

```
annotation(absoluteValue=false)
If false, then the variable defines a relative quantity, and if true an absolute quantity.
```
But when this model is exported to FMU, "relativeQuantity" attribute is not set:

```
<ScalarVariable name="Tdiff" valueReference="2">
<Real unit="K"/>
</ScalarVariable>
```
The expected behaviour will be:

```
<ScalarVariable name="Tdiff" valueReference="2">
<Real unit="K" relativeQuantity="true"/>
</ScalarVariable>
```

### Version and OS

 - OpenModelica Version: OpenModelica v1.24.0-dev-120-g4b5f3b42de (64-bit)
 - OS: e.g. Windows 10, 64 bit

**Comment by arun3688** [comment]
@casella sure i will look into this

**Comment by arun3688** [comment]
@BasilioPV can you provide a model with minimal example for the above mentioned problem, It will be easy to debug

**Comment by BasilioPV** [comment]
Yes, please find the varibale "Tdiff" on this model:

```
model eee
  Modelica.Units.SI.TemperatureDifference Tdiff;
  Modelica.Thermal.HeatTransfer.Components.HeatCapacitor heatCapacitor(C = 10000, T(start = 473.15, fixed = false))  annotation(
    Placement(transformation(origin = {-10, 38}, extent = {{-10, -10}, {10, 10}})));
  Modelica.Thermal.HeatTransfer.Components.ThermalConductor thermalConductor(G = 10)  annotation(
    Placement(transformation(origin = {10, 8}, extent = {{-10, -10}, {10, 10}})));
  Modelica.Thermal.HeatTransfer.Sources.FixedTemperature fixedTemperature(T = 303.15)  annotation(
    Placement(transformation(origin = {46, 8}, extent = {{10, -10}, {-10, 10}}, rotation = -0)));
equation
  connect(heatCapacitor.port, thermalConductor.port_a) annotation(
    Line(points = {{-10, 28}, {-10, 8}, {0, 8}}, color = {191, 0, 0}));
  connect(thermalConductor.port_b, fixedTemperature.port) annotation(
    Line(points = {{20, 8}, {36, 8}}, color = {191, 0, 0}));
  Tdiff=thermalConductor.port_a.T-thermalConductor.port_b.T;
  

annotation(version="1.0",
    uses(Modelica(version = "4.0.0")),
    Documentation(info = "<html><head></head><body>Text editor sample</body></html>", revisions = "<html><head></head><body>Revision text</body></html>"));

end eee;
```

**Comment by arun3688** [comment]
> ### Description
> In a model I have defined a temperature difference variable:
> 
> ```
> Modelica.Units.SI.TemperatureDifference Tdiff;
> ```
> 
> where:
> 
> ```
> type TemperatureDifference = Real (
>     final quantity="ThermodynamicTemperature",
>     final unit="K") annotation(absoluteValue=false);
> ```
> 
> and from Modelica Language Specification:
> 
> ```
> annotation(absoluteValue=false)
> If false, then the variable defines a relative quantity, and if true an absolute quantity.
> ```
> 
> But when this model is exported to FMU, "relativeQuantity" attribute is not set:
> 
> ```
> <ScalarVariable name="Tdiff" valueReference="2">
> <Real unit="K"/>
> </ScalarVariable>
> ```
> 
> The expected behaviour will be:
> 
> ```
> <ScalarVariable name="Tdiff" valueReference="2">
> <Real unit="K" relativeQuantity="true"/>
> </ScalarVariable>
> ```
> 
> ### Version and OS
> * OpenModelica Version: OpenModelica v1.24.0-dev-120-g4b5f3b42de (64-bit)
> * OS: e.g. Windows 10, 64 bit

**Comment by arun3688** [comment]
@BasilioPV What if the annotation of `absoluteValue` is set to true

```
annotation(absoluteValue=true )
If false, then the variable defines a relative quantity, and if true an absolute quantity.
```

what should be the expected behaviour in modeldescription.xml, should the `relativeQuantity` be set to false ? , I am little confused here

The expected behaviour will be:

<ScalarVariable name="Tdiff" valueReference="2">
<Real unit="K" relativeQuantity="false"/>
</ScalarVariable>

**Comment by BasilioPV** [comment]
@arun3688 '''absoluteValue''' is by default true, so usually '''absoluteValue=true''' sentence is not used.

**Comment by arun3688** [comment]
@BasilioPV The issue is fixed with this commit e87692cb9321fd3159d90c3e1275b29bc92bae8e, I tested with your example and now relativeQuantity is exported to fmu's see below

```
  <!-- Index of variable = "3" -->
  <ScalarVariable
    name="Tdiff"
    valueReference="2"
    >
    <Real unit="K" relativeQuantity="true"/>
  </ScalarVariable>
```
Please test it from the nightly builds tomorrow(6-7-2024)

**Comment by BasilioPV** [comment]
Thanks @arun3688 ! Work as expected!
